src/Symfony/Component/Worker/MessageFetcher/MessageFetcherInterface.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Worker\MessageFetcher;
+
+/**
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+interface MessageFetcherInterface
+{
+    /**
+     * @return string|bool The message or false
```

Comment:
`MessageCollection|bool The message collection or false` ?

Reply:
`fetchMessages` should always return a `MessageCollection`. The message collection might we empty, but that is okey. 
However, do not change the signature for the RouterInterface